crates/core/src/hash.rs

```diff
@@ -236,6 +244,11 @@ pub trait HashAlgorithm {
 
     /// Computes the hash of the provided data with a prefix.
     fn hash_prefixed(&self, prefix: &[u8], data: &[u8]) -> Hash;
+
+    /// Computes the hash of the provided blinded data.
+    fn hash_blinded(&self, data: &Blinded<Vec<u8>>) -> Hash {
```

Comment:
This was needed because currently when hashing the blinded plaintext, it gets serialized first, e.g. here
https://github.com/tlsnotary/tlsn/blob/38104bca1a12d7dc49b2fc62b74b0e29b4bb2572/crates/core/src/transcript/hash.rs#L65

This means that some serialization data will be present in the pre-image. Which means that my Poseidon hasher cannot reliably know the last 16 bytes of the pre-image is the blinder.

Recall that the Poseidon hasher places the blinder in a separate field element, so it must be able to separate it from the plaintext. (The need to put the blinder into a separate field element is a limitation of the halo2 circuit. This limitation can potentially be lifted but requires redesigning the circuit.)

Reply:
To summarize, are you suggesting that in this place
`if commitment.hash.value != alg.hash_canonical(&self.data) {`
we should check if the hash alg requires AuthDecode, and if so, then we use hash(data|blinder).
For all other algs, we keep using `hash_canonical(&self.data)`, correct?